src/classes/MyOtherClass.cls

```diff
@@ -5,7 +5,7 @@
 
 
 
-public class MyOtherClass {
+public without sharing class MyOtherClass {
```

Comment:
Class `MyOtherClass` does not enforce sharing correctly. Please change to `with sharing`.